add usertesting.com guide, slight restructure of research (#4806)
sourcegraph/about: handbook/product/design/research/index.md

There is a #hallway-testing Slack channel to share information and scout participants for hallway tests.

### Unmoderated user testing

When unmoderated user testing is valuable we use [usertesting.com](../tools/index.md#product-design-and-research) to conduct unmoderated user research with a pool of participants sourced by usertesting.com.

> [Read the Sourcegraph guide for conducting research with usertesting.com](./user-testing-com.md)

### Moderated user testing

When moderated user testing is valuable, we use [LookBack](../tools/index.md#product-design-and-research) to conduct sessions. With LookBack, other team members can observe and take notes in real time without appearing to the participant, which makes for a more positive participation experience.

### Competitive and analogous research

It can be beneficial to see how other tools and platforms are solving parallel problems. We usually collect and store screenshots and references in artifacts like RFCs and the Research layer in Figma projects. We also have [Styleboards in Figma](https://www.figma.com/files/project/10712517/Styleboards?fuid=1011662758768504201) for collecting inspiration for product areas.

> ### Existing user research

Author: ```suggestion
### Existing user feedback
```

As a design team, we carry out research in a variety of forms depending on our research goals. Sometimes this research takes place at the beginning of a project, during the discovery phase. Other times, it takes place at the end of a project after shipping a design outcome.

# Research methods

Some specific methods we use:

### Hallway testing

> When we want to test our ideas using interactive prototypes or even in the product itself, our own team members make great participants. (However, we need to remember that our own team members are generally very experienced in the area of code search, which results in implicit bias that we must account for in synthesis.)

Reviewer: ```suggestion
When we want to test our ideas using interactive prototypes or even in the product itself, our own team members make great participants. (However, we need to remember that our own team members are generally very experienced in the area of code search, which results in implicit bias that we must account for in synthesis.)

There is a #hallway-testing Slack channel to share information and scout participants for hallway tests. 
```

### Unmoderated user testing

When unmoderated user testing is valuable we use [usertesting.com](../tools/index.md#product-design-and-research) to conduct unmoderated user research with a pool of participants sourced by usertesting.com.

> [Read the Sourcegraph guide for conducting research with usertesting.com](./user-testing-com.md)

### Moderated user testing

When moderated user testing is valuable, we use [LookBack](../tools/index.md#product-design-and-research) to conduct sessions. With LookBack, other team members can observe and take notes in real time without appearing to the participant, which makes for a more positive participation experience.

### Competitive and analogous research

> It's worthwhile to see how other tools and platforms are solving parallel problems. We usually keep screenshots and references in artifacts like RFCs and in project Figma files.

Reviewer: ```suggestion
It can be beneficial to see how other tools and platforms are solving parallel problems. We usually collect and store screenshots and references in artifacts like RFCs and the Research layer in Figma projects. We also have [Styleboards in Figma](https://www.figma.com/files/project/10712517/Styleboards?fuid=1011662758768504201) for collecting inspiration for product areas. 
```

> ### Existing artifact analysis.

Reviewer: ```suggestion
### User Feedback
```

Author: I'm going to change this slightly to 'Existing user feedback'
